nautobot/core/tests/runner.py

```diff
@@ -69,3 +79,88 @@ def teardown_databases(self, old_config, **kwargs):
         print("Database emptied!")
 
         super().teardown_databases(old_config, **kwargs)
+
+    def generate_report(self, test_results, result):
+        """
+        Generate Performance Report consists of unittests that are significantly slower than baseline.
+        """
+        test_result_count = len(test_results)
+
+        # Add `--generate_report` to the end of `invoke` commands to generate a report.json file consist of the performance tests result
+        if self.report_path:
+            data = {
+                "tests": [{"name": func_name, "execution_time": float(timing)} for func_name, timing in test_results],
+                "test_count": result.testsRun,
+                "failed_count": len(result.errors + result.failures),
+                "total_execution_time": result.timeTaken,
+            }
+            with open(self.report_path, "w") as outfile:
+                json.dump(data, outfile)
+        # Print the results in the CLI.
+        else:
+            if test_result_count:
+                print(f"\n{test_result_count} abnormally slower tests:")
+                for func_name, timing in test_results:
+                    time = float(timing)
+                    baseline = float(self.baselines[func_name])
+                    print(f"{time:.4f}s {func_name} is significantly slower than the baseline {baseline:.4f}s")
+                # Make the test fail if there are tests significantly slower
+                assert test_result_count == 0, "Performance Tests failed due to significantly slower tests"
+
+            if not test_results:
+                print("\nNo tests signficantly slower than baseline. Success!")
+
+    def get_baselines(self):
+        """Load the performance_baselines.yml file for result comparison."""
+        baselines = {}
+        input_file = "nautobot/core/tests/performance_baselines.yml"
```

Comment:
This should probably be configurable, as in the future plugins may want to generate and check their own performance baselines using this tooling.